oscarapi/basket/operations.py

```diff
@@ -26,7 +26,7 @@
 def apply_offers(request, basket):
     "Apply offers and discounts to cart"
     if not basket.is_empty:
-        Applicator().apply(request, basket)
+        Applicator().apply(basket, request)
```

Comment:
I think you have been using oscar HEAD?

https://github.com/django-oscar/django-oscar/blob/1.0.2/oscar/apps/offer/utils.py#L21

That is 1.0.2 and clearly request comes first and then after that basket.

That is some ugly backwards compatibility if you ask me.